lang/java/reef-common/src/main/java/org/apache/reef/driver/evaluator/EvaluatorRequest.java

```diff
@@ -196,6 +209,12 @@ private Builder(final EvaluatorRequest request) {
       }
     }
 
+    @SuppressWarnings("checkstyle:hiddenfield")
+    public T setSchedulingConstraint(final SchedulingConstraint schedulingConstraint) {
```

Comment:
This assumes that there is only one scheduling constraint. While this is the case now, would we want this to be the case in the future? If not, it might be better to have a `Set` of them and replace this method with `addSchedulingConstraint`.